ci/containers/Containerfile.base

```diff
@@ -9,3 +9,4 @@ RUN python3 -m pip install --break-system-packages tox
 
 ENV PIP_EXTRA_INDEX_URL=https://download.pytorch.org/whl/cu128
 ENV CUDA_HOME="/usr/local/cuda"
+ENV PYTEST_XDIST_AUTO_NUM_WORKERS=12
```

Comment:
FWIW I think it's more ergonomic to set this for all users, to say 4 by default? I for one was not happy to have my CPU usage jump to 100% with `tox -e tests`

Reply:
We might want to do this for everything yes. Maybe to `max(12, num_cores * 0.8)`?

Reply:
Yes, that can be implemented generally as a pytest hook. (I assume you mean min, not max).